crowbar_framework/lib/crowbar/registry/barclamp.rb

```diff
@@ -0,0 +1,214 @@
+#
+# Copyright 2011-2013, Dell
+# Copyright 2013-2015, SUSE Linux GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# TODO(must): Replace user_managed attribute with listed
+# TODO(must): Alias deprecated display_name with dislay
+
+module Crowbar
+  class Registry
+    class Barclamp
+      delegate(
+        :name,
+        :display,
+        :description,
+        :member,
+        :requires,
+        :listed,
+        :hidden,
+        :layout,
+        :version,
+        :schema,
+        :order,
+        :run_order,
+        :chef_order,
+        :nav,
+        :members,
+        :category,
+        :member_of?,
+        :category_of?,
+        to: :class
+      )
+
+      def [](name)
+        send(name.to_sym)
+      end
+
+      def to_s
+        display
+      end
+
+      class << self
+        attr_reader :name
+        attr_reader :display
+        attr_reader :description
+
+        attr_reader :member
+        attr_reader :requires
+
+        attr_reader :listed
+
+        attr_reader :layout
+        attr_reader :version
+        attr_reader :schema
+
+        attr_reader :order
+        attr_reader :run_order
+        attr_reader :chef_order
+
+        attr_reader :nav
+
+        def name(value = nil)
+          if value.nil?
+            @name || ""
+          else
+            @name = value
+          end
+        end
+
+        def display(value = nil)
```

Comment:
Of course we can do something like that. And as it's not finished it's marked as WIP (and because I had to do something different before finishing it).